### PR TITLE
[duboku] Fix hostname change and require referer

### PR DESCRIPTION
**Pull request by mozbugbox**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible.

### CODE REVIEW

File: yt_dlp/extractor/duboku.py [hunk 1]
```diff
@@ -150,36 +161,29 @@ def _real_extract(self, url):
             'episode_number': int_or_none(episode_id),
             'episode_id': episode_id,
             'formats': formats,
-            'http_headers': {'Referer': 'https://www.duboku.co/static/player/videojs.html'}
+            'http_headers': {'Referer': 'https://w.duboku.io/'}
```
Comment (pukkandan) [inline yt_dlp/extractor/duboku.py:164]:
```suggestion
            'http_headers': headers
```
doesn't this work?

Comment (mozbugbox) [reply yt_dlp/extractor/duboku.py:164]: Yes, it does. I missed it.

File: yt_dlp/extractor/duboku.py [hunk 2]
```diff
@@ -51,31 +51,41 @@ def _get_element_by_tag_and_attrib(html, tag=None, attribute=None, value=None, e
...
+        'url': 'https://w.duboku.io/vodplay/1575-1-1.html',
         'info_dict': {
             'id': '1575-1-1',
-            'ext': 'ts',
+            'ext': 'mp4',
             'series': '白色月光',
             'title': 'contains:白色月光',
             'season_number': 1,
             'episode_number': 1,
+
```
Comment (pukkandan) [inline yt_dlp/extractor/duboku.py:66]:
```suggestion
```

File: yt_dlp/extractor/duboku.py [hunk 3]
```diff
@@ -51,31 +51,41 @@ def _get_element_by_tag_and_attrib(html, tag=None, attribute=None, value=None, e
...
         'info_dict': {
             'id': '1588-1-1',
-            'ext': 'ts',
+            'ext': 'mp4',
             'series': '亲爱的自己',
-            'title': 'contains:预告片',
+            'title': 'contains:第1集',
             'season_number': 1,
             'episode_number': 1,
+
```
Comment (pukkandan) [inline yt_dlp/extractor/duboku.py:84]:
```suggestion
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Pull request merged**